internal/interface/handlers/user_handler.go

```diff
@@ -50,7 +52,21 @@ func (h *UserHandler) GetUserByID(c *fiber.Ctx) error {
 // @Failure 500 {object} response.Response
 // @Router /users [post]
 func (h *UserHandler) CreateUser(c *fiber.Ctx) error {
-	return nil
+	var createUserDTO dtos.CreateUserDTO
+	if err := c.BodyParser(&createUserDTO); err != nil {
+		resp := response.NewResponseFactory(response.ERROR, err.Error())
+		return resp.SendResponse(c, fiber.StatusBadRequest)
+	}
+
+	var req dtos.UserDTO
+	err := h.userUsecase.CreateUser(&req, &createUserDTO)
+	if err != nil {
+		resp := response.NewResponseFactory(response.ERROR, err.Error())
+		return resp.SendResponse(c, fiber.StatusInternalServerError)
```

Comment:
@wiraphatys we always return 500 if svc error right? Can we return some kind of custom error that has status codes in usecase layer so that we can just get the status codes from svc error in handler layer so that we don't need to switch case svc error in handler layer

Reply:
do you mean create a new error package like 
```go
package apperror
type AppError struct {
	Name       string
	HttpCode int
}
func (e *AppError) Error() string {
	return e.Name
}
var (
	BadRequest         = &AppError{"Bad request", http.StatusBadRequest}
	Unauthorized       = &AppError{"Unauthorized", http.StatusUnauthorized}
        ...
)
func BadRequestError(message string) *AppError {
	return &AppError{message, http.StatusBadRequest}
}
...
```

Reply:
I agree with you kub bro, but maybe can just using switch case for the svc error in handler and then apply the statusCode here. how about this idea kub ? 55555555555

Reply:
I'm fine with that kub, but it would mean switch case for every endpoint right?

https://github.com/isd-sgcu/rpkm67-gateway/blob/dev/internal/pin/pin.handler.go
```go
res, appErr := h.svc.FindAll(req)
	if appErr != nil {
		h.log.Named("FindAll").Error("FindAll: ", zap.Error(appErr))
		c.ResponseError(appErr)
		return
	}
```
if svc already returns custom error with status code, you can just response `appErr` in handler. If it is not an error from svc call, we can return a response generated from factory kub.

Reply:
svc = usecase, sorry i forgot555

Reply:
so, we should create a custom context which wraps `*fiber.Ctx` right

Reply:
not really, I think only a custom error for use case is enough, then we add another `resp.SendResponse` method that can accept the custom error type, let me try it in a new branch